### DISCUSSION TITLE
FilePathChooser setting default file name

### DISCUSSION THREAD

**Discussion by AJ6CU** [alejandroautalan/pygubu-designer#183] [General]
Probably doing this wrong...

1. In Pygubu created a textvariable "foo" for a FilePathChooser. I also set up the binding for on change

2. I do a foo.set("myfilename") 

3. Go thru file selection process (select "bar").   (on change selection fires)

4. Then switch to another area of the UX. Then come back to FilePathChooser. But before I do that, I: 

foo.set("myfilename") again to get back to default file name.

5. I go thru the file selection process, and again select "bar"

5. However, the on change event doesn't fire this time.  My hypothesis is that the FilePathChooser is remembering "bar" from before and that the foo.set("myfilename")   doesn't change the internal file previously?

Tried looking at the widget.config(textvariable="x") option but didn't get vary far and started seeing empty file names when i used it so, definitely doing something wrong there...

Thanks

Mark

**Comment by alejandroautalan** [comment]
Hello Mark.
Yes, the widget remembers the last path used and based on that it triggers the event. I does not monitor the associated tk variable.
To setup the path, you use the configure method:

    pathchooser.configure(path="Initial path")

I modified the example based on your usage case. Let me know if this helps.

https://github.com/alejandroautalan/pygubu-designer/blob/0e8cf0bc2a19227486973e84b97f618f1d2e4b13/examples/pathchooserdemo/pathchooserdemo.py#L16-L54

Regards
Alejandro A.

**Reply by AJ6CU** [reply]
ThanKs! This should help. One more question, after I select a file, I want to remember the directory that was used so next time, the user doesn't have to navigate. So I do:

self.savedFilePathChooserWidget.config(initialdir=os.path.dirname(os.path.abspath(self.savedFilePathChooser.get())))

Should I use the path= in addition this this?   Would be happy if  I set the initial directory to the last one, and then did something llike path="" to force an onchange even when a file is selected.

Regards,

Mark

**Reply by alejandroautalan** [reply]
It should work as you expect.
Example:

https://github.com/alejandroautalan/pygubu-designer/blob/01a53a19ba38211b4119ed93c2bb5443d2b96212/examples/pathchooserdemo/pathchooserdemo.py#L49-L53

Regards,
Alejandro A.

**Comment by AJ6CU** [comment]
Perfect! Thank you.

Regards,

Mark
